packages/components/src/components/files/WebDav.ts

```diff
@@ -0,0 +1,151 @@
+
+        return webDavFile.merge({
+            isCollection: values.collection,
+            isLeaf: values.leaf,
+            createdBy: values.createdby,
+            created: values.creationdate,
+            lastModified: values.lastmodified,
+            downloadUrl: values.href ? values.href + '?contentDisposition=attachment' : undefined,
+            name: values.text,
+        }) as WebDavFile;
+    }
+}
+
+function getWebDavUrl(containerPath: string, directory?: string, createIntermediates?: boolean) {
+    let url =
+        ActionURL.getContextPath() +
+        '/_webdav' +
+        ActionURL.encodePath(containerPath) +
+        '/' +
+        encodeURIComponent('@files');
+
+    if (directory) {
+        url += '/' + directory;
+    }
+
+    if (createIntermediates) {
+        url += '?createIntermediates=' + createIntermediates;
+    }
+
+    return url;
+}
+
+export function getWebDavFiles(
+    containerPath: string,
+    directory?: string,
+    includeDirectories?: boolean
+): Promise<Map<string, WebDavFile>> {
+    return new Promise((resolve, reject) => {
+        const url = getWebDavUrl(containerPath, directory);
+
+        return Ajax.request({
+            url: url + '?method=JSON',
+            method: 'GET',
+            success: Utils.getCallbackWrapper(response => {
+                // Filter directories and create webdav files
+                const filteredFiles = response.files.reduce((filtered, file) => {
+                    if (includeDirectories || !file.collection) {
+                        return filtered.set(file.text, WebDavFile.create(file));
+                    }
+                }, Map<string, WebDavFile>());
+
+                resolve(filteredFiles);
+            }),
+            failure: Utils.getCallbackWrapper(
+                response => {
+                    console.error('Problem retrieving webDav files for container ' + containerPath);
+                    reject(response);
+                },
+                null,
+                false
+            ),
+        });
+    });
+}
+
+export function uploadWebDavFile(
+    file: File,
+    containerPath: string,
+    directory?: string,
+    createIntermediates?: boolean
+): Promise<string> {
+    return new Promise((resolve, reject) => {
+        const form = new FormData();
+        form.append('file', file);
+
+        const url = getWebDavUrl(containerPath, directory, createIntermediates);
+
+        Ajax.request({
+            url,
+            method: 'POST',
+            form,
+            success: Utils.getCallbackWrapper(() => {
+                resolve(file.name);
+            }),
+            failure: Utils.getCallbackWrapper(
+                () => {
+                    console.error('failure uploading file ' + file.name);
+                    reject(file.name);
+                },
+                null,
+                false
+            ),
+        });
+    });
+}
+
+export function uploadWebDavFiles(
```

Comment:
This method doesn't seem to be used anywhere here or in the other provenance branches. Is it necessary?

Reply:
It's not currently used.  Initially I thought it was but I already moved it over from biologics

Reply:
I'd rather we didn't check in unused code. We have a rich history of checking in unused code that is then never used. It leads to code that is either unnecessarily maintained because folks will update it to make sure it continues to work despite it never being used, or it rots and is never updated to remain compatible, because it's never used. Either way, it's unnecessary and we should remove it.